provider/gce/environ.go

```diff
@@ -267,18 +268,15 @@ func (env *environ) Bootstrap(ctx environs.BootstrapContext, callCtx context.Pro
 // Destroy shuts down all known machines and destroys the rest of the
 // known environment.
 func (env *environ) Destroy(ctx context.ProviderCallContext) error {
-	ports, err := env.IngressRules(ctx)
+	err := destroyEnv(env, ctx)
 	if err != nil {
 		return errors.Trace(err)
 	}
-
-	if len(ports) > 0 {
-		if err := env.ClosePorts(ctx, ports); err != nil {
```

Comment:
Right, so removing the entire firewall entity is what we should have done in the first place.